int32 curve_id
float32 T
float32[] cx
float32[] cy
float32[] cz
float32 fh
